### PR TITLE
Add text case changing functionality (Fixes #2390)

### LINKED ISSUE #2390: Change text case
# IMPORTANT
* App version: 2.12.5 (154)
* Android version/rom: 11
* Device: Ulefone Powerarmor 13 

## Explain the Problem
After typing text, would like to change text case. Currently impossible.
E.g.
- all letters case switch: CASE, case
- over character switch: CaSe, cAsE
- all words first letter switch: First Letter, first letter
- sentence first letter switch: First letter, first letter

## Steps to Reproduce
- type text
- mark (select) text or if not, change whole document text case
- try to change it's case
- cannot change as no options for that

## Log
Not appropriate currently.

### PR DESCRIPTION
**Pull request by zeuchie**
Improvement to the Edit menu to include ability to change case. Fixes issue #2390 "Change text case".

This pull request introduces the following changes:
1. Added a new "TextCasingUtils" class to handle text case transformations.
2. Updated "DocumentEditAndViewFragment.java" to handle the new text case changing actions.
3. Enhanced "HighlightingEditor.java" to implement text replacement for case changes.
4. Modified "document__edit__menu.xml" to include a new submenu for text case operations.
5. Created a new drawable "ic_format_text_case_black_24dp.xml" for the text case menu icon.
6. Added new strings in "strings.xml" for the UI elements related to case changing.

These changes allow users to perform the following text case operations from the Edit menu:
- Toggle Case (Toggle all text to Uppercase or Lowercase)
- Switch Case (Switch the text case of each character)
- Capitalize Words (Capitalize the first letter of each word)
- Capitalize Sentences (Capitalize the first letter of each sentence)

The new functionality respects the user's text selection, applying changes only to the selected text if a selection is active, or to the entire document if no text is selected. Please review and let me know if any changes or improvements are needed.

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -477,4 +477,9 @@ work. If not, see <https://creativecommons.org/publicdomain/zero/1.0/>.
     <string name="wont_save_min_length">Files with fewer than %d characters not saved automatically in order to prevent data loss.</string>
     <string name="advanced_filtering_help"><![CDATA[<small><a href="https://www.github.com/gsantner/markor/discussions/1940">Advanced filtering syntax</a></small>]]></string>
     <string name="filter">Filter</string>
+    <string name="text_case">Text Case</string>
+    <string name="toggle_case">Toggle Case</string>
```
Comment (gsantner) [inline app/src/main/res/values/strings.xml:481]: Thanks,

Suggestion, put a example at each like  AbC -> aBc, abc -> ABC

Comment (zeuchie) [reply app/src/main/res/values/strings.xml:481]: Hello, in which file should I add the examples? I can add them into TextCasingUtils.java, DocumentEditAndViewFragment.java, or HighlightingEditor.java. For example I can add the following comments to TextCasingUtils.java:
```
package net.gsantner.markor.util;

public class TextCasingUtils {
    public static String toggleCase(String text) { // Toggle all text to Uppercase or Lowercase. Ex: "Case" -> "CASE" -> "case"
        if (text.equals(text.toUpperCase())) {
            return text.toLowerCase();
        } else {
            return text.toUpperCase();
        }
    }

    public static String switchCase(String text) { // Switch the text case of each character. Ex: "CaSe" -> "cAsE"
        StringBuilder result = new StringBuilder();
        for (char c : text.toCharArray()) {
            if (Character.isUpperCase(c)) {
                result.append(Character.toLowerCase(c));
            } else if (Character.isLowerCase(c)) {
                result.append(Character.toUpperCase(c));
            } else {
                result.append(c);
            }
        }
        return result.toString();
    }

    public static String capitalizeWords(String text) { // Capitalize the first letter of each word. Ex: "this is a test." -> "This Is A Test.
        StringBuilder result = new StringBuilder();
        boolean capitalizeNext = true;
        for (char c : text.toCharArray()) {
            if (Character.isWhitespace(c)) {
                capitalizeNext = true;
                result.append(c);
            } else if (capitalizeNext) {
                result.append(Character.toUpperCase(c));
                capitalizeNext = false;
            } else {
                result.append(Character.toLowerCase(c));
            }
        }
        return result.toString();
    }

    public static String capitalizeSentences(String text) { // Capitalize the first letter of each sentence. Ex: "sentence 1. Sentence 2! sEntence 3? enD." -> "Sentence 1. Sentence 2! SEntence 3? EnD."
        StringBuilder result = new StringBuilder();
        boolean capitalizeNext = true;
        for (char c : text.toCharArray()) {
            if (c == '.' || c == '!' || c == '?') {
                capitalizeNext = true;
                result.append(c);
            } else if (Character.isWhitespace(c)) {
                result.append(c);
            } else if (capitalizeNext) {
                result.append(Character.toUpperCase(c));
                capitalizeNext = false;
            } else {
                result.append(c);
            }
        }
        return result.toString();
    }
}
```

Comment (gsantner) [reply app/src/main/res/values/strings.xml:481]: Display text (strings.xml)

Comment (zeuchie) [reply app/src/main/res/values/strings.xml]: Oh I see, thank you. I added an example to the display text of each text case now.

**Review by gsantner — APPROVED** [review]
Thank you!

File: app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java [hunk 2]
```diff
@@ -297,6 +299,60 @@ private int rowEnd(final int y) {
         return layout.getLineEnd(line);
     }
 
+    // Text-Casing
+    // ---------------------------------------------------------------------------------------------
+    public void toggleCase() {
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java:304]: All of this functionality should be in TextViewUtils IMO. 

A good bit of it already exists in TextViewUtils

**Pull request merged**